packages/std/src/coins.rs

```diff
@@ -206,6 +208,83 @@ impl Coins {
 
         Ok(())
     }
+
+    /// Returns an iterator over the coins.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use cosmwasm_std::{coin, Coin, Coins, Uint128};
+    /// let mut coins = Coins::default();
+    /// coins.add(coin(500, "uluna")).unwrap();
+    /// coins.add(coin(1000, "uatom")).unwrap();
+    /// let mut iterator = coins.iter();
+    ///
+    /// let uatom = iterator.next().unwrap();
+    /// assert_eq!(uatom.denom, "uatom");
+    /// assert_eq!(uatom.amount.u128(), 1000);
+    ///
+    /// let uluna = iterator.next().unwrap();
+    /// assert_eq!(uluna.denom, "uluna");
+    /// assert_eq!(uluna.amount.u128(), 500);
+    ///
+    /// assert_eq!(iterator.next(), None);
+    /// ```
+    pub fn iter(&self) -> CoinsIter<'_> {
+        CoinsIter(self.0.iter())
+    }
+}
+
+impl IntoIterator for Coins {
+    type Item = Coin;
+    type IntoIter = CoinsIntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        CoinsIntoIter(self.0.into_iter())
+    }
+}
+
+impl<'a> IntoIterator for &'a Coins {
+    type Item = &'a Coin;
+    type IntoIter = CoinsIter<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+#[derive(Debug)]
+pub struct CoinsIntoIter(alloc::collections::btree_map::IntoIter<String, Coin>);
+
+impl Iterator for CoinsIntoIter {
+    type Item = Coin;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.next().map(|(_, coin)| coin)
+    }
+}
+
+impl DoubleEndedIterator for CoinsIntoIter {
+    fn next_back(&mut self) -> Option<Self::Item> {
+        self.0.next_back().map(|(_, coin)| coin)
+    }
+}
+
+#[derive(Debug)]
+pub struct CoinsIter<'a>(alloc::collections::btree_map::Iter<'a, String, Coin>);
+
+impl<'a> Iterator for CoinsIter<'a> {
+    type Item = &'a Coin;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.next().map(|(_, coin)| coin)
+    }
+}
+
+impl<'a> DoubleEndedIterator for CoinsIter<'a> {
+    fn next_back(&mut self) -> Option<Self::Item> {
+        self.0.next_back().map(|(_, coin)| coin)
+    }
```

Comment:
Can we also implement https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html for both iterators?